File: apps/android/app/src/main/java/com/filo/app/Analytics.kt
package com.filo.app

import android.content.Context
import android.os.Bundle
import com.google.firebase.analytics.FirebaseAnalytics

object Analytics {
    private var firebase: FirebaseAnalytics? = null

    fun initialize(context: Context) {
        firebase = FirebaseAnalytics.getInstance(context.applicationContext)
    }

    fun track(name: String, params: Map<String, Any?> = emptyMap()) {
        val bundle = Bundle()
        params.forEach { (key, value) ->
            when (value) {
                is String -> bundle.putString(key, value)
                is Int -> bundle.putLong(key, value.toLong())
                is Long -> bundle.putLong(key, value)
                is Double -> bundle.putDouble(key, value)
                is Boolean -> bundle.putBoolean(key, value)
            }
        }
        firebase?.logEvent(name, bundle)
    }

    fun screen(name: String) {
        track(
            FirebaseAnalytics.Event.SCREEN_VIEW,
            mapOf(
                FirebaseAnalytics.Param.SCREEN_NAME to name,
                FirebaseAnalytics.Param.SCREEN_CLASS to "Filo",
            ),
        )
    }
}

File: apps/ios/Filo/Sources/Analytics.swift
import FirebaseAnalytics

enum FiloAnalytics {
    static func track(_ name: String, parameters: [String: Any] = [:]) {
        Analytics.logEvent(name, parameters: parameters)
    }

    static func screen(_ name: String) {
        track(AnalyticsEventScreenView, parameters: [
            AnalyticsParameterScreenName: name,
            AnalyticsParameterScreenClass: "Filo",
        ])
    }
}
